Feature/movie details

baraa0abd opened the pull request:
https://github.com/user-attachments/assets/6d825c26-cccb-4271-a813-e3f1923c2af9


https://github.com/user-attachments/assets/588eff62-3167-43b0-b684-33a1b34347f8

Alaakhaled101 committed:
add implementation for Preview

baraa0abd committed:
Merge remote-tracking branch 'origin/refactor/design-system' into feature/movie-details

# Conflicts:
#	designSystem/build.gradle.kts

baraa0abd committed:
Merge remote-tracking branch 'origin/refactor/design-system-and-presentation' into feature/movie-details

baraa0abd committed:
Merge branch 'refs/heads/develop' into feature/movie-details

baraa0abd committed:
Refactor: Remove unused demo screen and data class

This commit removes the `MovieDetailsDemoScreen.kt` file, which contained a composable function used for previewing the movie details screen with sample data.

The `MovieDetailsData` data class, previously used to structure this sample data, has also been removed from `MovieDetailsScreen.kt` as it is no longer needed. The preview functionality previously relying on this has been commented out.

baraa0abd committed:
Refactor: Update icon resource imports in MovieDetailsNavigationHeader

The resource imports for icons in `MovieDetailsNavigationHeader.kt` have been updated to use the direct R class import from the `designsystem` module (`com.madrid.designsystem.R`) instead of the fully qualified name (`com.madrid.designSystem.R`).

baraa0abd committed:
Refactor MovieDetailsNavigationHeader icons

- Removed background from clickable icons.
- Updated share icon.

baraa0abd committed:
Refactor: Improve ReviewCard and related components

- Implemented `Chips` composable in `MovieDetailsHeader`.
- Refined `ReviewCard` layout and styling:
    - Adjusted width and corner radius.
    - Improved alignment and spacing of elements.
    - Updated text styles.
- Created `ReviewScreen` composable to display a horizontal list of `ReviewCard`s.
- Renamed `ReviewsScreen.kt` to `ReviewsTopbar.kt` and updated its content.
- Added `androidx.compose.foundation:foundation-android` dependency.
- Removed unused preview from `MovieDetailsNavigationHeader`.

baraa0abd committed:
Refactor: Simplify MainActivity and update dependencies

- Removed `MainScreen` composable and its associated navigation logic (`navController`, `navBackStackEntry`, `currentRoute`, `currentDestination`, `CustomBottomBar`, `MovioNavGraph`).
- Introduced `MovieDetailsScreenPreview` composable in `MainActivity.kt` to directly display a movie details screen preview.
- Populated `MovieDetailsScreenPreview` with sample data for movie details, cast members, and poster.
- Updated `androidx.compose.foundation:foundation-android` dependency version in `libs.versions.toml` and added a duplicate entry with a different version.
- Added the new `foundation-android` dependency to `app/build.gradle.kts`.

baraa0abd committed:
Refactor: Update background color in MainActivity

The background color of the main content area in `MainActivity.kt` has been changed from `Theme.color.surfaces.onSurfaceVariant` to `Theme.color.surfaces.surfaceContainer`.

baraa0abd committed:
Refactor: Reimplement SimilarMoviesScreen

- Recreated the `SimilarMoviesScreen.kt` file.
- Introduced a `SimilarMovie` data class.
- Implemented `SimilarMoviesScreen` composable:
    - Displays a header with "Similar Movies" title and a "See all" button.
    - Utilizes a `LazyVerticalGrid` to show a list of `MovieCard`s.
    - Uses fake movie data for preview purposes.
- Created a `MovieCard` composable:
    - Displays movie poster, title, and rating.
    - Uses `FilteredImage` for image loading.
    - Includes a rating badge.
- Added a preview function `SimilarMoviesScreenPreview`.
- Updated `MainActivity.kt` to include `SimilarMoviesScreen` in the preview layout along with other review components.

baraa0abd committed:
Refactor: Update SimilarMovies to be a section and display horizontally

- Renamed `SimilarMoviesScreen` to `SimilarMoviesSection`.
- Changed the layout from `LazyVerticalGrid` to `LazyRow` to display movies horizontally.
- Updated `onBackClick` to `onSeeAllClick` in `SimilarMoviesSection` parameters.
- Adjusted image corner radius in `MovieCard` from 12dp to 8dp.
- Adjusted rating badge corner radius in `MovieCard` from 8dp to 4dp.
- Updated the preview function to `SimilarMoviesSectionPreview` and wrapped its content in a `Box` with background color.
- Updated `MainActivity` to use the new `SimilarMoviesSection` composable.

baraa0abd committed:
Refactor: Improve Reviews and Similar Movies sections in MovieDetailsScreen

- Introduced `SectionHeader` composable for consistent section titles with "See all" functionality.
- Integrated `ReviewScreen` into `MovieDetailsScreen` to display a horizontal list of `ReviewCard`s.
- `ReviewScreen` now includes a header for "Reviews" and "Similar Movies" sections, both utilizing `SectionHeader`.
- Added `SimilarMoviesSection` (placeholder) to `ReviewScreen`.
- Adjusted `ReviewCard` height to `137.dp`.
- Updated `MovieDetailsScreen` to pass callbacks for "See all" actions to `ReviewScreen`.
- Changed background color of `MovieDetailsScreen`'s root `Box` to `surfaceContainer`.
- Removed `ReviewsTopbar.kt` and its associated composables as its functionality is now incorporated into `ReviewScreen` and `SectionHeader`.

baraa0abd committed:
Refactor: Simplify `ReviewScreen` composable

- Removed comments that stated the obvious functionality of the code.
- Removed the "Similar Movies" section, including its header, spacer, and the `SimilarMoviesSection` composable call.
- The `ReviewScreen` now solely focuses on displaying the reviews section.

baraa0abd committed:
Refactor: Enhance MovieDetailsScreen and update MovieDetailsHeader

- Updated `MovieDetailsHeader.kt`:
    - Removed a comment in `Image` composable regarding accessibility description.
- Updated `MovieDetailsScreen.kt`:
    - Utilized `TextWithReadMore` composable from the `designSystem` module to display the movie description, allowing for expandable text.
    - Set `maxLines` to 3 for the description.
    - Removed a local `TextWithReadMore` composable as it's now imported.

baraa0abd committed:
Refactor: Update MovieDetailsNavigationHeader icons

- Removed background from clickable icons.
- Updated share icon.

baraa0abd committed:
Refactor: Move `MovieDetailsNavigationHeader` to `MainActivity.kt` and integrate navigation

- The `MovieDetailsNavigationHeader` composable function, previously in its own file, has been moved into `MainActivity.kt`.
- `MainActivity.kt` now sets up the main screen content with `MovioTheme` and a `MainScreen` composable.
- `MainScreen` implements navigation using `rememberNavController` and `MovioNavGraph`.
- A `CustomBottomBar` is added for navigation between different destinations defined in `navBarDestinations`.
- `LocalNavController` is provided through `CompositionLocalProvider` to make the NavController accessible within the `MovioNavGraph`.
- The `enableEdgeToEdge()` function is called in `MainActivity.onCreate` for an immersive display experience.
- The preview function for `MovieDetailsNavigationHeader` has been removed.

ahmedhgabr commented on presentation/src/main/java/com/madrid/presentation/screens/detailsScreen/similarMovies/SimilarMoviesScreen.kt:
```diff
@@ -6,116 +6,172 @@ import androidx.compose.foundation.layout.Arrangement
 ) {
-    val state by viewModel.state.collectAsState()
-    var selectedMovieId by remember { mutableStateOf<String?>(null) }
+    val fakeMovies = listOf(
```
Remove the fake from the composable and use it in the preview

baraa0abd committed:
Refactor: Improve UI consistency and localization

- Replaced `Image` with `MovioIcon` in `MovieDetailsHeader` for consistency.
- Standardized the "See all" text using string resources in `SimilarMoviesScreen` and `SectionHeader`.
- Utilized string resources for the "Reviews" title in `ReviewScreen`.
- Replaced the star character with a `MovioIcon` in `SimilarMoviesScreen` for rating display.
- Added a new string resource `reviews` for "Reviews" and its Arabic translation.

baraa0abd committed:
Refactor: Replace AsyncImage with FilteredImage in ReviewCard

- Updated `ReviewCard.kt` to use the `FilteredImage` composable instead of `AsyncImage` for displaying reviewer images. This likely incorporates image filtering capabilities.
- Removed commented-out code and unused comments in `SimilarMoviesScreen.kt` for better readability.

baraa0abd committed:
Refactor: Update SimilarMoviesSection to accept movie list

- Modified `SimilarMoviesSection` composable to accept a `List<SimilarMovie>` as a parameter instead of using hardcoded fake data.
- Updated the preview function to pass the fake movie list to `SimilarMoviesSection`.

yasserahmed10 commented on presentation/src/main/java/com/madrid/presentation/component/MovieDetailsNavigationHeader.kt:
```diff
@@ -107,17 +94,3 @@ fun MovieDetailsNavigationHeader(
         }
     }
 }
-
```
Keep Preview here

yasserahmed10 commented on presentation/src/main/java/com/madrid/presentation/component/MovieDetailsNavigationHeader.kt:
Remove unused imports

yasserahmed10 commented on presentation/src/main/java/com/madrid/presentation/component/header/MovieDetailsHeader.kt:
```diff
@@ -72,7 +78,27 @@ fun MovieDetailsHeader(
+        )
+        MovioText(
+            text = text,
+            color = Theme.color.surfaces.onSurfaceVariant,
```
Change Color here to Theme.color.surfaces.onSurfaceContainer

yasserahmed10 commented on presentation/src/main/java/com/madrid/presentation/component/header/MovieDetailsHeader.kt:
```diff
@@ -72,7 +78,27 @@ fun MovieDetailsHeader(
+        MovioText(
+            text = text,
+            color = Theme.color.surfaces.onSurfaceVariant,
+            textStyle = Theme.textStyle.label.smallRegular14
```
Change text style here to Theme.textStyle.label.smallRegular12

yasserahmed10 commented on presentation/src/main/java/com/madrid/presentation/component/header/MovieDetailsHeader.kt:
```diff
@@ -72,7 +78,27 @@ fun MovieDetailsHeader(
+        modifier = Modifier
+            .background(
+                color = Theme.color.surfaces.surfaceContainer,
+                shape = RoundedCornerShape(16.dp)
```
Radius here 24.dp

baraa0abd committed:
Refactor: Update MovieDetailsHeader and NavigationHeader components

- **MovieDetailsHeader:**
    - Modified the `Chips` composable:
        - Increased corner radius to `24.dp`.
        - Updated text color to `Theme.color.surfaces.onSurfaceContainer`.
        - Changed text style to `Theme.textStyle.label.smallRegular12`.
- **MovieDetailsNavigationHeader:**
    - Removed unused imports:
        - `androidx.compose.foundation.background`
        - `androidx.compose.ui.tooling.preview.Preview`
        - `com.madrid.designSystem.theme.MovioTheme`

baraa0abd committed:
Test: Add Preview for MovieDetailsNavigationHeader

A new Composable function, `MovieDetailsNavigationHeaderPreview`, has been added. This function is annotated with `@Preview(showBackground = true, showSystemUi = true)` and `@Composable` to enable previewing the `MovieDetailsNavigationHeader` component in Android Studio.

Mohammed-qmr commented on presentation/src/main/java/com/madrid/presentation/component/header/MovieDetailsHeader.kt:
```diff
@@ -72,7 +78,27 @@ fun MovieDetailsHeader(
+                color = Theme.color.surfaces.surfaceContainer,
+                shape = RoundedCornerShape(24.dp)
+            )
+            .padding(horizontal = 8.dp, vertical = 4.dp)
```
U can use spacing

yasserahmed10 commented on presentation/src/main/java/com/madrid/presentation/screens/detailsScreen/detailsMovieScreen/MovieDetailsScreen.kt:
```diff
@@ -92,7 +83,8 @@ fun MovieDetailsScreen(
             TextWithReadMore(
                 description = uiState.description,
-                modifier = Modifier.padding(horizontal = 16.dp, vertical = 8.dp)
+                modifier = Modifier.padding(horizontal = 16.dp, vertical = 8.dp),
```
@Mohammed-qmr 
I have removed spacing and raduis from design system

Mohamedragabali reviewed (approved):
super work

baraa0abd committed:
Refactor: MovieDetailsScreen and related components

- **MovieDetailsScreen:**
    - Refactored `MovieDetailsScreen` to be a standalone composable.
    - Added `LaunchedEffect` to load data when the screen is composed.
    - Updated `TopAppBar` to be displayed within a `Box` with status bar padding.
    - Integrated `ExpandableDescription` for movie descriptions.
    - Updated `TopCastSection` to display cast members from the view model state.
    - Updated `ReviewScreen` to display reviews from the view model state.
    - Integrated `SimilarMoviesSection` to display similar movies from the view model state.
    - Made `loadData` function in `DetailsMovieViewModel` internal.

- **ExpandableDescription:**
    - Created a new composable `ExpandableDescription` to display text with a "Read More/Less" button.

- **TopAppBar:**
    - Added `onFirstIconClick`, `onSecondIconClick`, and `onThirdIconClick` parameters for icon click handling.
    - Added `isFavorite` parameter to control the appearance of the favorite icon.
    - Updated icon content descriptions.
    - Ensured a `Box` with the size of an icon is rendered if `firstIcon` is null to maintain layout.

- **ReviewScreen:**
    - Modified to accept `ReviewsScreenUiState` to display reviews.
    - Added an empty state message when no reviews are available.
    - Localized "No Reviews yet!" string.

- **MovioNavHost:**
    - Removed commented-out code related to `MovieDetailsScreen` navigation.

- **Strings:**
    - Added `no_reviews_available` string resource for English and Arabic.

- **Removed File:**
    - Deleted the old `MovieDetailsScreen.kt` file.

baraa0abd committed:
Merge remote-tracking branch 'origin/develop' into feature/movie-details

# Conflicts:
#	designSystem/src/main/java/com/madrid/designSystem/component/TopAppBar.kt
#	presentation/src/main/java/com/madrid/presentation/component/header/MovieDetailsHeader.kt
#	presentation/src/main/java/com/madrid/presentation/screens/detailsScreen/detailsMovieScreen/MovieDetailsScreen.kt
#	presentation/src/main/java/com/madrid/presentation/screens/detailsScreen/similarMovies/SimilarMoviesScreen.kt

AmrNasserSaad commented:
resolve conflicts and merge

baraa0abd committed:
Refactor: Adjust padding in SimilarMoviesScreen and update MovieDetailScreen

- **SimilarMoviesScreen:**
    - Changed horizontal padding in the `Row` from `16.dp` to `4.dp`.

- **MovieDetailScreen:**
    - Removed unused import `com.madrid.designSystem.component.MovioText`.
    - Removed commented-out code in `onMovieClick` lambda within `SimilarMoviesSection`.
